silkworm/silkrpc/commands/ots_api.cpp

```diff
@@ -498,6 +498,45 @@ boost::asio::awaitable<void> OtsRpcApi::handle_ots_getContractCreator(const nloh
     co_return;
 }
 
+boost::asio::awaitable<void> OtsRpcApi::handle_ots_traceTransaction(const nlohmann::json& request, nlohmann::json& reply) {
+    const auto& params = request["params"];
+    if (params.size() != 1) {
+        const auto error_msg = "invalid ots_traceTransaction params: " + params.dump();
+        SILK_ERROR << error_msg << "\n";
+        reply = make_json_error(request["id"], 100, error_msg);
+        co_return;
+    }
+
+    const auto transaction_hash = params[0].get<evmc::bytes32>();
+
+    SILK_DEBUG << "transaction_hash: " << transaction_hash << "\n";
+
+    auto tx = co_await database_->begin();
+
+    try {
+        ethdb::TransactionDatabase tx_database{*tx};
+        trace::TraceCallExecutor executor{io_context_, *block_cache_, tx_database, workers_};
+
+        const auto transaction_with_block = co_await core::read_transaction_by_hash(*block_cache_, tx_database, transaction_hash);
```

Comment:
`read_transaction_by_hash` return type is `std::optional<TransactionWithBlock>` because such function returns `std::nullopt` if the given `transaction_hash` (which is an input received from a client) is unknown or non-existent. So you *must* check `transaction_with_block` before using it:

```
if (not transaction_with_block) {
    // some code to log a warn and fill error response here... 
    co_await tx->close();
    co_return;
}
```